src/PlaneWaveBasis.jl

```diff
@@ -33,15 +33,15 @@ Discretization information for kpoint-dependent quantities such as orbitals.
 More generally, a kpoint is a block of the Hamiltonian;
 eg collinear spin is treated by doubling the number of kpoints.
 """
-struct Kpoint{T <: Real}
-    spin::Symbol                  # :up, :down, :both or :spinless
-    coordinate::Vec3{T}           # Fractional coordinate of k-Point
-    mapping::Vector{Int}          # Index of G_vectors[i] on the FFT grid:
-                                  # G_vectors(basis)[kpt.mapping[i]] == G_vectors(kpt)[i]
-    mapping_inv::Dict{Int, Int}   # Inverse of `mapping`:
-                                  # G_vectors(basis)[i] = G_vectors(kpt)[kpt.mapping_inv[i]]
-    G_vectors::Vector{Vec3{Int}}  # Wave vectors in integer coordinates:
-                                  # ({G, 1/2 |k+G|^2 ≤ Ecut})
+struct Kpoint
+    spin::Symbol                       # :up, :down, :both or :spinless
+    coordinate::Vec3{Rational{Int}}    # Fractional coordinate of k-Point
+    mapping::Vector{Int}               # Index of G_vectors[i] on the FFT grid:
+                                       # G_vectors(basis)[kpt.mapping[i]] == G_vectors(kpt)[i]
+    mapping_inv::Dict{Int, Int}        # Inverse of `mapping`:
+                                       # G_vectors(basis)[i] = G_vectors(kpt)[kpt.mapping_inv[i]]
+    G_vectors::Vector{Vec3{Int}}       # Wave vectors in integer coordinates:
+                                       # ({G, 1/2 |k+G|^2 ≤ Ecut})
```

Comment:
I think I only would have two spaces of space between the `Vec3{Rational{Int}}` and the first comment to avoid the lines getting too long :smile: